tools/Azure.Mcp.Tools.Speech/src/Services/SpeechService.cs

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Mcp.Core.Options;
+using Azure.Mcp.Core.Services.Azure;
+using Azure.Mcp.Core.Services.Azure.Tenant;
+using Azure.Mcp.Tools.Speech.Models;
+using Microsoft.CognitiveServices.Speech;
+using Microsoft.CognitiveServices.Speech.Audio;
+using Microsoft.Extensions.Logging;
+using SdkSpeechRecognitionResult = Microsoft.CognitiveServices.Speech.SpeechRecognitionResult;
+
+namespace Azure.Mcp.Tools.Speech.Services;
+
+public class SpeechService(ITenantService tenantService, ILogger<SpeechService> logger) : BaseAzureService(tenantService), ISpeechService
+{
+    private readonly ILogger<SpeechService> _logger = logger;
+    /// <summary>
+    /// Recognizes speech from an audio file using Azure AI Services Speech.
+    /// </summary>
+    /// <param name="endpoint">Azure AI Services endpoint (e.g., https://your-service.cognitiveservices.azure.com/)</param>
+    /// <param name="filePath">Path to the audio file to process</param>
+    /// <param name="language">Speech recognition language (default: en-US)</param>
+    /// <param name="phrases">Optional phrases to improve recognition accuracy</param>
+    /// <param name="format">Output format (simple or detailed)</param>
+    /// <param name="profanity">Profanity filtering option (masked, removed, or raw)</param>
+    /// <param name="retryPolicy">Optional retry policy for resilience</param>
+    /// <returns>Speech recognition result containing recognized text and metadata</returns>
+    public async Task<Models.SpeechRecognitionResult> RecognizeSpeechFromFile(
+        string endpoint,
+        string filePath,
+        string? language = null,
+        string[]? phrases = null,
+        string? format = null,
+        string? profanity = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        ValidateRequiredParameters(endpoint, filePath);
+
+        if (!File.Exists(filePath))
+        {
+            throw new FileNotFoundException($"Audio file not found: {filePath}");
```

Comment:
I think we'll run into permission issues - will the user have to put in the same path as the project in vscode?  What tests have been done for this?